src/app/GitCommands/Git/GitModule.cs

```diff
@@ -3232,6 +3234,11 @@ public IEnumerable<IObjectGitItem> GetTree(ObjectId? commitId, bool full, string
 
             ExecutionResult result = _gitExecutable.Execute(args, cache: isArtificial ? null : GitCommandCache, cancellationToken: cancellationToken);
 
+            if (isArtificial && !GitVersion.SupportUpdateRefs)
+            {
+                return _gitTreeParser.Parse(result.StandardOutput);
+            }
+
             return _gitTreeParser.Parse(result.StandardOutput);
```

Comment:
I do not see a difference between `then` and `else`.
`return _gitTreeParser.ParseLsFiles(result.StandardOutput);` instead?

Reply:
Good catch...
Merge error, this is no longer an error, just empty tree